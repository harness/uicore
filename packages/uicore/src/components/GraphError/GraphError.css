/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.main {
  margin-left: auto !important;
  margin-right: auto !important;
  text-align: center;
  .img {
    width: 100%;
    height: 100%;
  }
  .title {
    margin: 24px 0px;
  }
  .secondLink {
    margin-left: 24px;
  }
}
