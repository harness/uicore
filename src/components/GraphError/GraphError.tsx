/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import css from './GraphError.css'
import { Icon } from '../../icons/Icon'
import { Container } from '../Container/Container'
import { Link } from '../Link/Link'

export interface Props {
  image?: any
  width?: string
  height?: string
  title?: string
  linkText?: string
  link?: string
  onLinkClick?: () => void
  secondLinkText?: string
  secondLink?: string
  onSecondLinkClick?: () => void
}

function GraphError(props: Props) {
  return (
    <Container className={css.main}>
      {props.image ? (
        <Container width={props.width || 320} height={props.height || 180}>
          {props.image ? <img src={props.image} className={css.img} /> : <Icon name={'error'} size={120} />}
        </Container>
      ) : (
        <Container>
          <Icon name={'error'} size={120} />
        </Container>
      )}
      <Container className={css.title}>
        <strong>{props.title ? props.title : 'Error, This Provider is not showing any data '}</strong>
      </Container>
      {props.linkText && (props.onLinkClick || props.link) ? (
        <span className={css.link}>
          {props.onLinkClick ? (
            <a
              onClick={() => {
                props.onLinkClick!()
              }}>
              {props.linkText}
            </a>
          ) : (
            <Link target="_blank" href={props.link}>
              {props.linkText}
            </Link>
          )}
        </span>
      ) : null}

      {props.secondLinkText && (props.onSecondLinkClick || props.secondLink) ? (
        <span className={css.secondLink}>
          {props.onSecondLinkClick ? (
            <a
              onClick={() => {
                props.onSecondLinkClick!()
              }}>
              {props.secondLinkText}
            </a>
          ) : (
            <Link target="_blank" href={props.link}>
              {props.secondLinkText}
            </Link>
          )}
        </span>
      ) : null}
    </Container>
  )
}

export { GraphError }
