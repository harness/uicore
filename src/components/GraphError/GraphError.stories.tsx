/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable no-alert */
import React from 'react'
import type { Meta, Story } from '@storybook/react'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary } from '@storybook/addon-docs/blocks'
import { GraphError, Layout } from '../..'
import { Props as GraphErrorProps } from '../GraphError/GraphError'

export default {
  title: 'Components / GraphError',

  component: GraphError,
  parameters: {
    layout: 'centered',
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>GraphError</Title>
            <Subtitle>
              <pre>
                <code>{`import {GraphError} from '@harness/uicore'`}</code>
              </pre>
            </Subtitle>

            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  },
  decorators: [Story => <Story />]
} as Meta
export const Basic: Story<GraphErrorProps> = args => {
  return (
    <>
      <Layout.Horizontal spacing="small">
        <GraphError
          linkText={'View in Splunk'}
          onLinkClick={() => {
            alert('clicked')
          }}
          secondLinkText={'View call Logs'}
          onSecondLinkClick={() => {
            alert('clicked')
          }}
          {...args}></GraphError>
      </Layout.Horizontal>
    </>
  )
}
