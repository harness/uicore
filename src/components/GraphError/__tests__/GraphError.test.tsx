import React from 'react'
import { render, wait } from '@testing-library/react'
import { GraphError } from '../GraphError'

describe('GraphError unit tests', () => {
  test('Render GraphError with title and link should match snapshot', async () => {
    const { container } = render(
      <GraphError linkText={'View in Splunk'} onLinkClick={() => void 0} title={'Error occured'} />
    )
    await wait()
    expect(container).toMatchSnapshot()
  })

  test('Render GraphError with title and link should render title and link', async () => {
    const { container } = render(
      <GraphError linkText={'View in Splunk'} onLinkClick={() => void 0} title={'Error occured'} />
    )
    await wait()
    expect(container.querySelector('[class*="title"]')).not.toBeNull()
    expect(container.querySelector('svg')).not.toBeNull()
    expect(container.querySelector('a[href]')).toBeNull()
  })

  test('Render GraphError with title, image and no link should render image, title and link without href', async () => {
    const { container } = render(
      <GraphError
        linkText={'View in Splunk'}
        image={'someimg'}
        link={'https://www.google.com'}
        title={'Error occured'}
      />
    )
    await wait()
    expect(container.querySelector('[class*="title"]')).not.toBeNull()
    expect(container.querySelector('img')).not.toBeNull()
    expect(container.querySelector('svg')).toBeNull()
    expect(container.querySelector('a[href]')).not.toBeNull()
  })
})
