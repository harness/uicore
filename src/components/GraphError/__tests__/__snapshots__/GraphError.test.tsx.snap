// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphError unit tests Render GraphError with title and link should match snapshot 1`] = `
<div>
  <div
    class="font main"
  >
    <div
      class="font main"
    >
      <span
        aria-hidden="true"
        class="bp3-icon bp3-icon-error main"
        icon="error"
      >
        <svg
          data-icon="error"
          height="120"
          viewBox="0 0 20 20"
          width="120"
        >
          <path
            d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 16H9v-2h2v2zm0-3H9V4h2v9z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="font main title"
    >
      <strong>
        Error occured
      </strong>
    </div>
    <span
      class="link"
    >
      <a>
        View in Splunk
      </a>
    </span>
  </div>
</div>
`;
